camptix.php

```diff
@@ -2791,43 +2791,33 @@ function generate_attendee_report( $format ) {
 			$report = '<?xml version="1.0" encoding="UTF-8"?>' . PHP_EOL . '<attendees>' . PHP_EOL;
 
 		$paged = 1;
+		$buyer_map = array();
+
+		// Ordering by ID ASC is important. Presence of buyer row depends on it. Buyer has to fetched, same or before rest of attendees rows are fetched.
 		while ( $attendees = get_posts( array(
 			'post_type' => 'tix_attendee',
 			'post_status' => array( 'publish', 'pending' ),
-			'posts_per_page' => 200,
+			'posts_per_page' => 500,
 			'paged' => $paged++,
 			'orderby' => 'ID',
 			'order' => 'ASC',
-			'cache_results' => false,
 		) ) ) {
 
-			$attendee_ids = array();
-			foreach ( $attendees as $attendee )
-				$attendee_ids[] = $attendee->ID;
-
-			/**
-			 * Magic here, to by-pass object caching. See Revenue report for more info.
-			 */
-			$this->filter_post_meta = $this->prepare_metadata_for( $attendee_ids );
-			unset( $attendee_ids, $attendee );
-
 			foreach ( $attendees as $attendee ) {
 				$attendee_id = $attendee->ID;
 
-				$buyer = get_posts( array(
-					'post_type'      => 'tix_attendee',
-					'post_status'    => array( 'publish', 'pending' ),
-					'posts_per_page' => 1,
-					'orderby'        => 'ID',
-					'order'          => 'ASC',
+				$access_token = get_post_meta( $attendee->ID, 'tix_access_token', true );
 
-					'meta_query'     => array(
-						array(
-							'key'    => 'tix_access_token',
-							'value'  => get_post_meta( $attendee->ID, 'tix_access_token', true ),
-						),
-					),
-				) );
+				/**
+				 So when a buyer buys tickets for a bunch of attendees, access token for all those attendees would be same.
+				 Further, the buyer will always be inserted first into the database.
+				 Which means that when access token for a lot attendees is same, we can figure out the buyer by finding the first attendee with same access token.
+				 * **/
```

Comment:
Why does GitHub show this in red? Oh, I bet it's because it has a PHPDoc `/**` comment instead of a regular `/*` one.